services/applications/olex2_linux/prebuild__01_create_olex2_headless.py

```diff
@@ -0,0 +1,461 @@
+    <cmd
+      <cmd6 "spy.runODAC('spy.ac_run solveit -%1')">
+    >
+  >
+>
+
+
+<ac6 help="Launches ac_run"
+  <body
+    <args
+      <arg1 name="single" def="">
+    >
+    <cmd
+      <cmd1 "spy.ac6.auto()">
+    >
+  >
+>
+
+<start_autochem help="Launches ac_run"
+  <body
+    <args
+      <arg1 name="filefull" def="'FileFull()'">
+    >
+    <cmd
+      <cmd1 "reap %1.ins">
+      <cmd2 "spy.ac6.auto()">
+    >
+  >
+>
+
+<start_rpac help="Launches ReportPlusAC"
+  <body
+    <args
+      <arg1 name="filefull" def="'FileFull()'">
+    >
+    <cmd
+      <cmd1 "reap %1.ins">
+      <cmd2 "spy.ac6.create_report()">
+    >
+  >
+>
+
+<onexit help="Executes on program exit"
+  <body
+  <cmd
+    <cmd "stop logging">
+    <cmd "flush history">
+    <cmd "if IsFileLoaded() then 'spy.saveHistory()>>spy.SaveStructureParams()>>
+          spy.SaveCifInfo()'">
+    <cmd "spy.SaveUserParams()">
+    <cmd "spy.SaveOlex2Params()">
+    <cmd "spy.threading.joinAll()">
+    <cmd "spy.onexit()">
+    <cmd "echo Good Bye">
+  >
+ >
+>
+
+<reapcif help="Load a Crystallographic File"
+  <body
+    <args
+      <arg1 name="file name" def="">
+      <arg2 name="options" def="">
+      <arg3 name="width" def="100">
+      <arg4 name="filename" def="filename()">
+      <arg5 name="base" def="basedir()">
+      <arg6 name="ext" def="fileext()">
+      >
+    <cmd
+      <cmd1  "if strcmp(%4,'none') then none else spy.saveHistory()">
+      <cmd2  "spy.SaveStructureParams()">
+      <cmd4  "@reap %1">
+      <cmd4  "echo 'loading'">
+      <cmd9  "spy.OnStructureLoaded(%4)">
+      <cmd4  "echo 'loaded'">
+      <cmd7  "SetVar(snum_refinement_sg,sg(%n))">
+      <cmd8  a="spy.SetParam(snum.refinement_max_peaks,Ins(QNUM))">
+      <cmd9  a="spy.SetParam(snum.refinement_max_cycles,Ins(LS))">
+      <cmd10 "if File.Exists(strdir()/filename().vvd) then none else 'SetVar(snum_refinement_original_formula,xf.GetFormula())'">
+      <cmd11 "if and(strcmp(%4,filename()),strcmp(%6,fileext())) then none else 'spy.loadHistory()'">
+      <cmd14 cmd1="if strcmp(%4,filename()) then none else 'clear'">
+    >
+  >
+>
+
+<export help="Exports data from CIF"
+  <body
+    <args
+      <arg1 name="hkl_name" def="">
+    >
+    <cmd
+      <cmd1 "@export %1">
+      <cmd2 "spy.cif.reloadMetadata(True)">
+    >
+  >
+>
+
+<#include "custom.xld">
+<#include "app.ConfigDir()/custom.xld">
+"""
+def create_new_file():
+    logger.debug(f'Downloading Olex2 archive from: {url}')
+    r = requests.get(url, timeout=600)
+    zip_io = io.BytesIO(r.content)
+    zip_file = zipfile.ZipFile(zip_io, 'a')
+    zip_file.writestr('olex2/startc', startc_str)
+    zip_file.writestr('olex2/macrox.xld', macrox_string)
+    zip_file.write('olex2_files/olex2c-linux64', 'olex2/olex2c-linux64')
+    zip_file.close()
+    with open(output_path, 'wb') as fobj:
+        fobj.write(zip_io.getbuffer())
+
+def main():
+    if not output_path.exists():
+        if not pathlib.Path('olex2_files/olex2c-linux64'):
```

Comment:
```suggestion
        if not pathlib.Path('olex2_files/olex2c-linux64').exists():
```
As it stands I think this line always evaluates to `True`. I assume it is meant to check for the existence of the file?